packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -724,7 +725,9 @@ function generateResolver({ generatorOptions, metadata }: { generatorOptions: Cr
 
         @Query(() => ${metadata.className})
         @SubjectEntity(${metadata.className})
-        async ${instanceNameSingular}(@Args("id", { type: () => ID }) id: string): Promise<${metadata.className}> {
+        async ${instanceNameSingular}(@Args("id", { type: () => ID }) id: ${
+        integerTypes.includes(metadata.properties.id.type) ? "number" : "string"
```

Comment:
id is a string at runtime, you can't change that using types only - there needs to be some sort of conversion going on

Reply:
I tested this at runtime with a query like this:
`query MyQuery($id: ID!) {
  myResolver(id: $id) {
        id
  }
}`

and arguments like this:
`{ id: 1 }`

and it worked without a conversion.

Reply:
please veryify the type of id at runtime.

(mikro-orm accepts number and string)

Reply:
@nsams id was actually a string at runtime, so I added a transformer and updated the code.